jujugui/static/gui/src/app/store/env/go.js

```diff
@@ -476,25 +476,32 @@ YUI.add('juju-env-go', function(Y) {
     handleEnvironmentInfo: function(data) {
       if (data.Error) {
         console.warn('Error retrieving environment information.');
+        return;
+      }
+      var self = this;
+      // Store default series and provider type in the env.
+      var response = data.Response;
+      self.set('defaultSeries', response.DefaultSeries);
+      self.set('providerType', response.ProviderType);
+      if (localStorage.getItem('environmentName')) {
+        self.set('environmentName', localStorage.getItem('environmentName'));
       } else {
-        // Store default series and provider type in the env.
-        var response = data.Response;
-        this.set('defaultSeries', response.DefaultSeries);
-        this.set('providerType', response.ProviderType);
-        if (localStorage.getItem('environmentName')) {
-          this.set('environmentName', localStorage.getItem('environmentName'));
-        } else {
-          this.set('environmentName', response.Name);
-        }
-        // For now we only need to call environmentGet if the provider is MAAS.
-        if (response.ProviderType === 'maas') {
-          this.environmentGet();
-        } else {
-          // Set the MAAS server to null, so that subscribers waiting for this
-          // attribute to be set can be released.
-          this.set('maasServer', null);
-        }
+        self.set('environmentName', response.Name);
+      }
+      // For now we only need to call environmentGet if the provider is MAAS.
+      if (response.ProviderType !== 'maas') {
+        // Set the MAAS server to null, so that subscribers waiting for this
+        // attribute to be set can be released.
+        self.set('maasServer', null);
+        return;
       }
+      self.environmentGet(function(data) {
```

Comment:
If you switch to an arrow function here you don't need to do the `var self = this` above. 

```JavaScript
this.environmentGet((data) => {
  ...
}
```